src/com/machinepublishers/jbrowserdriver/ContextItem.java

```diff
@@ -154,16 +154,27 @@ void init(final JBrowserDriverServer driver, final Context context) {
         Util.handleException(e);
       }
       final ContextItem thisObject = this;
-      AppThread.exec(context.statusCode, new Sync<Object>() {
-        @Override
-        public Object perform() {
-          engine.get().setJavaScriptEnabled(SettingsManager.settings().javascript());
-          httpListener.set(new HttpListener(thisObject,
-              context.statusCode, context.timeouts.get().getPageLoadTimeoutObjMS()));
-          Accessor.getPageFor(view.get().getEngine()).addLoadListenerClient(httpListener.get());
-          engine.get().setCreatePopupHandler(new PopupHandler(driver, context));
-          return null;
+      AppThread.exec(context.statusCode, () -> {
+        Settings settings = SettingsManager.settings();
```

Comment:
ContextItem essentially corresponds to a browser window, and each shares cookies/etc. So I think local storage should be shared across ContextItems. You could move most of this to the Context constructor, but here still set the userDataDir on the WebEngine.

Also in Context#reset it could delete all the files with the dir if it's not a custom dir specified by the user. `FileUtils.cleanDirectory(` would be useful for that.